src/main/java/org/opentripplanner/graph_builder/module/GtfsModule.java

```diff
@@ -181,6 +227,39 @@ private GtfsMutableRelationalDao loadBundle(GtfsBundle gtfsBundle)
 
         GtfsFeedId gtfsFeedId = gtfsBundle.getFeedId();
 
+        // check for duplicate bundle names and make sure that at most one of those bundles contains an auto-generated
+        // feed ID. If there is more than one, non-deterministic Feed IDs could be generated.
+        String bundleFilename = gtfsBundle.getPath().getName();
+        if (bundleFilenamesWithAutoGeneratedFeedId.contains(bundleFilename)) {
+            throw new RuntimeException(
+                String.format(
+                    "There are at least 2 bundles with the name `%s` that lack a feed_id value in a record in the feed_info.txt table. This will result in non-deterministic Feed ID generation!",
+                    bundleFilename
+                )
+            );
+        }
+        if (gtfsFeedId.isAutoGenerated()) {
+            bundleFilenamesWithAutoGeneratedFeedId.add(bundleFilename);
+            autoGeneratedFeedIds.add(gtfsFeedId.getId());
+        }
+
+        // Read the agency table to determine if there is a single agency entry without an agency ID. This is used later
+        // to determine whether an agency ID was auto-generated. According to the GTFS spec, if there are multiple
+        // agencies, then they must have an agency ID. Therefore, it is only necessary to read the first record.
+        boolean agencyFileContainedOneAgencyWithoutId = false;
+        InputStream agencyInputStream = gtfsBundle.getCsvInputSource().getResource("agency.txt");
```

Comment:
It would be useful to note in the comment above where the agencyId actually gets auto-generated. Is this in the OBA code referenced below (`OBA replaces it with the agency's name`)?